src/cr/cube/matrix/assembler.py

```diff
@@ -821,6 +831,20 @@ def _derived_element_idxs(self, dimension, order):
             if derived
         )
 
+    def _diff_element_idxs(self, dimension, order):
+        """Return tuple(int) of difference elements' indices for a dimension."""
+        return tuple(
+            element_index
+            for element_index, is_difference in enumerate(
+                np.array(
+                    # ---Valid elements are not differences
+                    [False for _ in dimension.valid_elements]
+                    + [element.is_difference for element in dimension.subtotals]
+                )[order]
+            )
+            if is_difference
+        )
```

Comment:
IMHO i think that this logic should be divided into who belongs in the dimension and who in the assembler. I would propose that this property becomes:
```python
    def _diff_element_idxs(self, dimension, order):
        """Return tuple(int) of difference elements' indices for a dimension."""
        return tuple(np.where(dimension.subdiff_map[order])[0])
```

Where `subdiff_map` is this property within the `Dimension` class in the `dimension.py` module

```python
    @lazyproperty
    def subdiff_map(self) -> np.ndarray:
        return np.array(
            # ---Valid elements are not differences
            [False for _ in self.valid_elements]
            + [subtotal.is_difference for subtotal in self.subtotals]
        )
```
What do you think?

Reply:
Well, nobody is going to use the `subdiff_map`, other than this property. So I'd say that it's unearned abstraction. I do get your point, but I'd still keep it this way. Maybe I can simplify the expression a little bit.

Reply:
ok maybe you can use the `np.where` instead of the external iteration